system_metrics_collector/src/system_metrics_collector/linux_cpu_measurement_node.cpp

```diff
@@ -117,4 +118,36 @@ system_metrics_collector::ProcCpuData LinuxCpuMeasurementNode::makeSingleMeasure
   }
 }
 
+void LinuxCpuMeasurementNode::publishStatistics()
+{
+  if (publisher_ == nullptr) {
+    return;
+  }
+
+  const moving_average_statistics::StatisticData statistic_data = getStatisticsResults();
+  const double data[moving_average_statistics::STATISTICS_DATA_TYPES.size()] = {
+    statistic_data.average,
+    statistic_data.min,
+    statistic_data.max,
+    statistic_data.standard_deviation,
+    static_cast<double>(statistic_data.sample_count)
+  };
+
+  MetricsMessage msg = newMetricsMessage();
+  msg.metrics_source = "cpu_usage";
+  for (int i = 0; i < moving_average_statistics::STATISTICS_DATA_TYPES.size(); ++i) {
+    msg.statistics.emplace_back();
+    msg.statistics.back().data_type = moving_average_statistics::STATISTICS_DATA_TYPES[i];
+    msg.statistics.back().data = data[i];
```

Comment:
:warning: you are making the assumption that `data[i]` and `STATISTICS_DATA_TYPES` have the same size

Reply:
`data` was initialized to be the size of `STATISTICS_DATA_TYPES`.

Anyway, this kind of code is no longer part of this PR.